torchx/test/torchx_test.exs

```diff
@@ -0,0 +1,144 @@
+defmodule TorchxTest do
+  use ExUnit.Case, async: true
+
+  doctest Torchx
+
+  alias Torchx.Backend, as: TB
+
+  # Torch Tensor creation shortcut
+  defp tt(data, type \\ {:f, 32}), do: Nx.tensor(data, type: type, backend: TB)
+
+  @types [{:s, 8}, {:u, 8}, {:s, 16}, {:s, 32}, {:s, 64}, {:bf, 16}, {:f, 32}, {:f, 64}]
+  @bf16_and_ints [{:s, 8}, {:u, 8}, {:s, 16}, {:s, 32}, {:s, 64}, {:bf, 16}]
+  @ints [{:s, 8}, {:u, 8}, {:s, 16}, {:s, 32}, {:s, 64}]
+  @ops [:add, :subtract, :divide, :remainder, :multiply, :power, :atan2, :min, :max]
+  @bitwise_ops [:bitwise_and, :bitwise_or, :bitwise_xor, :left_shift, :right_shift]
+  @logical_ops [:equal, :not_equal, :greater, :less, :greater_equal, :less_equal, :logical_and, :logical_or, :logical_xor]
+  @unary_ops [:abs, :bitwise_not, :ceil, :floor, :negate, :round, :sign]
+
+  defp test_binary_op(op, data_a \\ [[1, 2], [3, 4]], data_b \\ [[5, 6], [7, 8]], type_a, type_b) do
+    a = tt(data_a, type_a)
+    b = tt(data_b, type_b)
+
+    c = Kernel.apply(Nx, op, [a, b])
+
+    binary_a = Nx.backend_transfer(a, Nx.BinaryBackend)
+    binary_b = Nx.backend_transfer(b, Nx.BinaryBackend)
+    binary_c = Kernel.apply(Nx, op, [binary_a, binary_b])
+
+    assert(Nx.backend_transfer(c) == binary_c)
+  end
+
+  defp test_unary_op(op, data \\ [[1, 2], [3, 4]], type) do
+    t = tt(data, type)
+
+    r = Kernel.apply(Nx, op, [t])
+
+    binary_t = Nx.backend_transfer(t, Nx.BinaryBackend)
+    binary_r = Kernel.apply(Nx, op, [binary_t])
+
+    assert(Nx.backend_transfer(r) == binary_r)
+  end
+
+  describe "binary tensor-tensor ops" do
+    for op <- @ops ++ @logical_ops,
+        type_a <- @types,
+        type_b <- @types,
+        not ((op in [:divide, :remainder, :atan2, :power] and (type_a == {:bf, 16} and type_b in @bf16_and_ints)) or
+               (type_b == {:bf, 16} and type_a in @bf16_and_ints)) do
```

Comment:
Instead of having the `not`, break them in subgroups: has a module attribute with only `[:divide, :remainder, :atan2, :power]` so you can check these rules.